lighthouse-core/computed/metrics/lantern-largest-contentful-paint.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('../computed-artifact.js');
+const LanternMetric = require('./lantern-metric.js');
+const LHError = require('../../lib/lh-error.js');
+const LanternFirstContentfulPaint = require('./lantern-first-contentful-paint.js');
+
+/** @typedef {import('../../lib/dependency-graph/base-node.js').Node} Node */
+
+class LanternLargestContentfulPaint extends LanternMetric {
+  /**
+   * @return {LH.Gatherer.Simulation.MetricCoefficients}
+   */
+  static get COEFFICIENTS() {
+    return {
+      intercept: 0,
+      optimistic: 0.5,
+      pessimistic: 0.5,
+    };
+  }
+
+  /**
+   * TODO: Validate.
+   * @param {Node} dependencyGraph
+   * @param {LH.Artifacts.TraceOfTab} traceOfTab
+   * @return {Node}
+   */
+  static getOptimisticGraph(dependencyGraph, traceOfTab) {
+    const lcp = traceOfTab.timestamps.largestContentfulPaint;
+    if (!lcp) {
+      throw new LHError(LHError.errors.NO_LCP);
+    }
+
+    return LanternFirstContentfulPaint.getFirstPaintBasedGraph(
+      dependencyGraph,
+      lcp,
+      _ => true
```

Comment:
I'm still working on this.

Reply:
Haven't made any progress on improving this. @patrickhulce any ideas? Or maybe the accuracy is good enough (would want to use the new data you're collecting from GCP before saying for sure).

Reply:
yeah let's wait for the full dataset before we invest too much more time.

@connorjclark I'll need to ping you separately for a more privileged key to run the actual collection since mine is capped at 200/day

https://github.com/GoogleChrome/lighthouse/tree/patrick_collect_gcp/lighthouse-core/scripts/lantern/collect

if you want to peek though :)